### ISSUE TITLE
Pulumi requires databrick's managed resource group to exist while Azure requires the opposite

### ISSUE THREAD

**Issue by thiDucTran** [pulumi/pulumi-azure-native#3643]
### What happened?

according to https://www.pulumi.com/registry/packages/azure-native/api-docs/databricks/workspace/#managedresourcegroupid_go , it is implied that the input of managed resource is required and also said resource group must exist ..but creating databricks would fail with the message in Azure

```json
"statusMessage": "{\"status\":\"Failed\",\"error\":{\"code\":\"ResourceOperationFailure\",\"message\":\"The resource operation completed with terminal provisioning state 'Failed'.\",\"details\":[{\"code\":\"ApplianceProvisioningFailed\",\"message\":\"The managed resource group '/subscriptions/foo/resourceGroups/mrg-bar' already exists. Please specify a new non-existent managed resource group\"}]}}",
```

### Example

```go
    	createManagedResourceGroup, _ := resources.NewResourceGroup(ctx, "managedResourceGroup", &resources.ResourceGroupArgs{
			Location:          pulumi.String(location),
			ResourceGroupName: pulumi.String("m"+resourceGroupName),
		},pulumi.DependsOn([]pulumi.Resource{createResourceGroup}))
```

```go
		_ , err := databricks.NewWorkspace(ctx, "databricks", &databricks.WorkspaceArgs{
			Location:               pulumi.String(location),
			ManagedResourceGroupId: createManagedResourceGroup.ID(),
			ResourceGroupName: pulumi.String(resourceGroupName),
			WorkspaceName: pulumi.String("databricks-dev"),
			RequiredNsgRules: pulumi.String("AllRules"),
			Sku: &databricks.SkuArgs{
				Name: pulumi.String("premium"),
			},
			Parameters: &databricks.WorkspaceCustomParametersArgs{
				CustomPrivateSubnetName: &databricks.WorkspaceCustomStringParameterArgs{
					Value: pulumi.String("databricks-container-subnet"),
				},
				CustomPublicSubnetName: &databricks.WorkspaceCustomStringParameterArgs{
					Value: pulumi.String("databricks-host-subnet"),
				},
				CustomVirtualNetworkId: &databricks.WorkspaceCustomStringParameterArgs{
					Value: createVnet.ID(),
				},
				EnableNoPublicIp: &databricks.WorkspaceCustomBooleanParameterArgs{
					Value: pulumi.Bool(true),
				},
			},
		},pulumi.DependsOn([]pulumi.Resource{createVnet}),pulumi.DependsOn([]pulumi.Resource{createHostSubnet}),pulumi.DependsOn([]pulumi.Resource{createContainerSubnet}))
		if err != nil {
			return err
		}
```

### Output of `pulumi about`

```
CLI
Version      3.136.1
Go Version   go1.23.2
Go Compiler  gc

Plugins
KIND      NAME          VERSION
resource  azure-native  2.64.2
resource  azure-native  2.64.2
resource  azure-native  2.64.2
resource  azure-native  2.42.1
resource  azure-native  2.42.1
language  go            unknown

Host
OS       ubuntu
Version  22.04
Arch     x86_64
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by thiDucTran** [comment]
pulumi does not require the managed resource group to be pre-existing before creating databricks..which makes sense..I just have the wrong assumption since the input https://www.pulumi.com/registry/packages/azure-native/api-docs/databricks/workspace/#managedresourcegroupid_go needs to be provided in Azure's resource ID format.

TLDR: dont create the managed resource group..only provide its name in the format like below
`ManagedResourceGroupId: pulumi.String("/subscriptions/"+subscription+"/resourceGroups/m"+resourceGroupName),`
